Userland/Libraries/LibHTTP/HttpRequest.cpp

```diff
@@ -12,33 +12,33 @@
 
 namespace HTTP {
 
-DeprecatedString to_deprecated_string(HttpRequest::Method method)
+StringView to_deprecated_string(HttpRequest::Method method)
 {
     switch (method) {
     case HttpRequest::Method::GET:
-        return "GET";
+        return "GET"sv;
     case HttpRequest::Method::HEAD:
-        return "HEAD";
+        return "HEAD"sv;
     case HttpRequest::Method::POST:
-        return "POST";
+        return "POST"sv;
     case HttpRequest::Method::DELETE:
-        return "DELETE";
+        return "DELETE"sv;
     case HttpRequest::Method::PATCH:
-        return "PATCH";
+        return "PATCH"sv;
     case HttpRequest::Method::OPTIONS:
-        return "OPTIONS";
+        return "OPTIONS"sv;
     case HttpRequest::Method::TRACE:
-        return "TRACE";
+        return "TRACE"sv;
     case HttpRequest::Method::CONNECT:
-        return "CONNECT";
+        return "CONNECT"sv;
     case HttpRequest::Method::PUT:
-        return "PUT";
+        return "PUT"sv;
     default:
         VERIFY_NOT_REACHED();
     }
 }
 
-DeprecatedString HttpRequest::method_name() const
+StringView HttpRequest::method_name() const
 {
     return to_deprecated_string(m_method);
```

Comment:
I got confused for a second here thinking we were returning a view to a temporary. Perhaps should rename this to_deprecated_string method?

Reply:
Absolutely, that's super confusing!
This PR is made out of an old branch, it was probably named `to_string()` back then.